seeding.js

```diff
@@ -0,0 +1,221 @@
+  `Swag slow-carb narwhal farm-to-table bicycle rights hashtag. Beard franzen venmo, keytar hella fixie plaid edison bulb vice. Pork belly unicorn banh mi, pickled keffiyeh humblebrag fam copper mug try-hard hell of cliche biodiesel.`,
+  `Keytar bitters gentrify vinyl. Poke aesthetic selfies lo-fi 8-bit. Af salvia pork belly chartreuse snackwave. Gastropub raclette hell of crucifix. Quinoa la croix copper mug photo booth crucifix shoreditch single-origin coffee four dollar toast hammock subway tile swag. Four loko shoreditch microdosing, tattooed drinking vinegar man bun vice sustainable normcore health goth ramps DIY.`,
+  `Neutra shaman ramps, lyft food truck tumeric keytar etsy asymmetrical authentic chambray butcher. Ugh mlkshk etsy typewriter tumblr dreamcatcher, af copper mug.`,
+  `Roof party palo santo single-origin coffee, tattooed chillwave retro cliche shabby chic jianbing pug bespoke. YOLO sriracha kombucha church-key iceland bicycle rights semiotics butcher tbh before they sold out schlitz.`
+];
+
+var reviewText = () => randReview[Math.floor(Math.random() * (randReview.length))];
+
+var randRestaurantId = ()=>restaurantsJson[Math.floor(Math.random() * (restaurantsJson.length))].ID;
+
+var randUserId = () => Math.floor(Math.random() * 50) + 1; //will be creating 50 users
+
+//generate a ISO 8601 date
+var randDate = () => {
+  var years = [2015, 2016, 2017, 2018];
+  var randYear = years[Math.floor(Math.random() * (years.length))];
+  var randMonth = Math.floor(Math.random() * 12) + 1;
+  var randMonthLength = randMonth === 2 ? 28 : [1, 3, 5, 7, 8, 10, 12].includes(randMonth) ? 31 : 30;
+  randMonth = randMonth < 10 ? `0${randMonth}` : randMonth;
+  var randDay = Math.floor(Math.random() * randMonthLength) + 1;
+  randDay = randDay < 10 ? `0${randDay}` : randDay;
+
+  return `${randYear}-${randMonth}-${randDay}`;
+};
+
+//generate tags
+var tagOptions = [
+  'Authentic',
+  'Bar Seating', 'Book the Bar',
+  'Cellar', 'Charming', 'Check the Wait', 'Comfort Food', 'Convenient', 'Couples', 'Cozy', 'Craft Beer Selection',
+  'Downstairs', 'Dancing',
+  'Family Style', 'Farm to Table', 'Fit for Foodies', 'Full Bar',
+  'Good Value', 'Good for Birthdays', 'Good for a Date', 'Great for Brunch',
+  'Healthy', 'Historic', 'Hot Spot',
+  'Kid-friendly',
+  'Live Music', 'Local Ingredients',
+  'Neighborhood Gem',
+  'Open Kitchen', 'Organic',
+  'Prix Fixe Menu',
+  'Quick Bite',
+  'Scenic View', 'Seasonal', 'Special Occasion',
+  'Tapas', 'Tasting Menu',
+  'Vegan', 'Vibrant Bar Scene',
+  'Waterfront', 'Wood Oven'
+];
+
+var randTags = () => {
+  var tagNumber = Math.floor(Math.random() * 10) + 1; //determine a number of tags a review will have
+  var selectedTags = [];
+  for (var i = 0; i < tagNumber; i++) {
+    var findTag = tagOptions[Math.floor(Math.random() * (tagOptions.length))];
+    if (selectedTags.indexOf(findTag) === -1) {
+      selectedTags.push(findTag);
+    }
+  }
```

Comment:
Checking if a tag is in the "selectedTags" array is a linear operation executed many times. 

It's more efficient to have this for loop build into an object as objects are hash maps. you won't be able to set duplicate keys.

then, when the for loop is complete return only the object keys.